### PR TITLE
Add utility to install CustomScript extension on VMs

### PR DESCRIPTION
**Pull request by lizzha**
This PR adds a script to install CustomScript on VMs within an Azure subscription, it supports to install on
1. single VM in a resource group
2. all VMs in a resource group
3. all VMs in a subscription

### CODE REVIEW

File: Utilities/Install-CustomScript.ps1 [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	-StorageAccountKey, the key of the storage account that contains the custom scripts
+	-OSType, the type of the OS, Linux or Windows, valid only if neither VmName nor ResourceGroupName is provided
+
+.NOTES
+	Creation Date:
+	Purpose/Change:
+
+.EXAMPLE
+	.\Utilities\Install-CustomScript.ps1 -AzureSecretsFile <PathToSecretFile> -VmName <VmName> -ResourceGroupName <RgName> `
+		-FileUris "https://teststorageaccout.blob.core.windows.net/script/install.sh" -CommandToRun "bash install.sh" `
```
Comment (chvalean) [inline Utilities/Install-CustomScript.ps1:25]: scripts should not reside on a location without versioning.
That is bad for code maintenance and tracking of changes.

Comment (lizzha) [reply Utilities/Install-CustomScript.ps1:25]: agree, this is for installing security packages on all VMs in our subscription, to meet the security coverage, the installation script is put in Azure storage as it contains internal packages, I will add version to the internal script

File: Utilities/Install-CustomScript.ps1 [hunk 2]
```diff
@@ -0,0 +1,177 @@
+##############################################################################################
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+#
+# Install-CustomScript.ps1
+<#
+.SYNOPSIS
+
+.PARAMETER
+	-AzureSecretsFile, the path of Azure secrets file
```
Comment (juhlee-microsoft) [inline Utilities/Install-CustomScript.ps1:10]: @lizzha how can you test in all possible scenario? Also add description in PR.

File: Utilities/Install-CustomScript.ps1 [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	[String] $VmName = "",
+	[String] $ResourceGroupName = "",
+	[Parameter(Mandatory=$true)]
+	[String] $FileUris,
+	[Parameter(Mandatory=$true)]
+	[String] $CommandToRun,
+	[String] $StorageAccountName = "",
+	[String] $StorageAccountKey = "",
+	[ValidateSet('Linux','Windows', IgnoreCase = $false)]
+	[String] $OSType = "Linux"
```
Comment (juhlee-microsoft) [inline Utilities/Install-CustomScript.ps1:43]: no WSL?

File: Utilities/Install-CustomScript.ps1 [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+	Get-ChildItem .\Libraries -Recurse | Where-Object { $_.FullName.EndsWith(".psm1") } | ForEach-Object { Import-Module $_.FullName -Force -Global -DisableNameChecking }
+	if (!$global:LogFileName){
+		Set-Variable -Name LogFileName -Value $LogFileName -Scope Global -Force
+	}
+
+	#Read secrets file and terminate if not present.
+	if ($AzureSecretsFile)
+	{
+		$secretsFile = $AzureSecretsFile
+	}
```
Comment (juhlee-microsoft) [inline Utilities/Install-CustomScript.ps1]: Please keep the consistent coding style like "} else { "

File: Utilities/Install-CustomScript.ps1 [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+		} catch {
+			Write-LogErr "Exception occurred in when installing CustomScript extension on VM $($vm.Name)."
+			Write-LogErr $_.Exception
+		}
+	}
+	$jobs | Wait-Job -Timeout 360
+	foreach ($job in $jobs) {
+		$state = $job.State
+		if ($state -eq "Running") {
+			$state = "Timeout"
```
Comment (juhlee-microsoft) [inline Utilities/Install-CustomScript.ps1:142]: You did not put timeout-ed VM in the failed count. Does it mean we only count VM in power-off state if it fails?

**Review by juhlee-microsoft — CHANGES_REQUESTED** [review]

File: Utilities/Install-CustomScript.ps1 (hunk 3 above)
Comment (lizzha) [reply Utilities/Install-CustomScript.ps1:43]: It's the "Operation System" field of the VM displayed in Azure portal. Now it contains only "Linux" and "Windows"

File: Utilities/Install-CustomScript.ps1 (hunk 2 above)
Comment (lizzha) [reply Utilities/Install-CustomScript.ps1:10]: I tested for the below scenarios, for both Linux and Windows VMs
1. single VM in a resource group
2. all VMs in a resource group
3. all VMs in a subscription

File: Utilities/Install-CustomScript.ps1 (hunk 4 above)
Comment (lizzha) [reply Utilities/Install-CustomScript.ps1]: will update

File: Utilities/Install-CustomScript.ps1 (hunk 5 above)
Comment (lizzha) [reply Utilities/Install-CustomScript.ps1:142]: will include the timeout VM in the failed count.

Comment (lizzha) [reply Utilities/Install-CustomScript.ps1:142]: updated

**Review by juhlee-microsoft — APPROVED** [review]

**Pull request merged**